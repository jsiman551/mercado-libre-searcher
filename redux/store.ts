import { configureStore } from '@reduxjs/toolkit'
import searchDataReducer from './slices/search-data-slice'
import searchInputReducer from './slices/search-input-slice'
import sortOptionReducer from './slices/sort-option-slice'
import sidebarFlagReducer from './slices/sidebar-flag-slice'
import priceRangeReducer from './slices/price-range-slice'

export const store = configureStore({
  reducer: {
    searchData: searchDataReducer,
    searchInput: searchInputReducer,
    sortOption: sortOptionReducer,
    mobileSidebarFlag: sidebarFlagReducer,
    priceRange: priceRangeReducer,
  },
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
