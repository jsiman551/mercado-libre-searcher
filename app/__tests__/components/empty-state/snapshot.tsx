import { render } from '@testing-library/react'
import EmptyState from '@/components/empty-state'
import { store } from '@/redux/store'
import { Provider } from 'react-redux'

it('renders EmptyState component', () => {
  const { container } = render(
    <Provider store={store}>
      <EmptyState />
    </Provider>,
  )
  expect(container).toMatchSnapshot()
})
