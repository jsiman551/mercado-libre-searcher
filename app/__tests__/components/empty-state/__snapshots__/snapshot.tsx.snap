// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders EmptyState component 1`] = `
<div>
  <div
    class="sc-beyTiQ TyDoB"
    data-testid="emptyResultsContainer"
  >
    <p
      class="sc-guDMob kbsjPd"
    >
      Intenta realizar una b√∫squeda para visualizar una lista de productos.
    </p>
  </div>
</div>
`;
